src/coreclr/vm/memberload.h

```diff
@@ -62,10 +62,10 @@ class MemberLoader
 public:
     static void DECLSPEC_NORETURN ThrowMissingMethodException(MethodTable* pMT,
                                             LPCSTR szMember,
-                                            ModuleBase *pModule,
-                                            PCCOR_SIGNATURE pSig,
-                                            DWORD cSig,
-                                            const SigTypeContext *pTypeContext);
+                                            ModuleBase *pModule = NULL,
```

Comment:
The implementation does not deal with null default values well. If any of these arguments are null, the output is going to degrade to a generic message.

For example, try:
```
using System.Runtime.CompilerServices;

unsafe class Test
{
    static void Main()
    {
        F(default, null);
    }

    [UnsafeAccessorAttribute(UnsafeAccessorKind.StaticMethod)]
    extern static void F(Test _, delegate* unmanaged[Cdecl]<void> pfn);

    static void F(void* pfn)
    {
    }
}
```
It is going to print:
```
Unhandled exception. System.MissingMethodException: Method not found: '?'.
   at Test.F(Test _,  pfn)
   at Test.Main()
```

Reply:
The same repro fails with the following in the checked builds of the AOT compiler:
```
 Generating native code
  Unhandled Exception: EETypeRva:0x009F30D0(System.Diagnostics.DebugProvider+DebugAssertException):    at System.Diagno
  stics.DebugProvider.Fail(String, String) + 0x37
     at System.Diagnostics.Debug.Fail(String, String) + 0x32
     at Internal.IL.UnsafeAccessors.GenerateAccessorSpecificFailure(UnsafeAccessors.GenerationContext&, String, Boolean
  ) + 0x2af
     at Internal.IL.UnsafeAccessors.TryGetIL(EcmaMethod) + 0x29f
     at Internal.IL.NativeAotILProvider.GetMethodIL(MethodDesc) + 0x9b
     at ILCompiler.FeatureSwitchManager.GetMethodIL(MethodDesc) + 0x3c
     at ILCompiler.Compilation.CombinedILProvider.GetMethodIL(MethodDesc) + 0x1a
```

Reply:
Ugh. I didn't look in detail at the impl of `MemberLoader::ThrowMissingMethodException()`. It is rather unexpected given the contract at the top of the function. I will fix it up so it works in a more intuitive manner.

> The same repro fails with the following in the checked builds of the AOT compiler:

Hmmm. That is odd.

Reply:
> > The same repro fails with the following in the checked builds of the AOT compiler:

> Hmmm. That is odd.

The runtime tests do not run on native AOT by default. They need to be run manually. I would expect this problem to be caught by the tests.

Reply:
I have been running them manually and it didn't trigger. I build them Debug with no issues. I'll try with the above example.